subsys/net/ip/tcp.c

```diff
@@ -1936,17 +1936,21 @@ NET_CONN_CB(tcp_established)
 	u8_t tcp_flags;
 	u16_t data_len;
 
+	k_mutex_lock(&context->lock, K_FOREVER);
+
 	NET_ASSERT(context && context->tcp);
 
 	tcp_hdr = net_tcp_get_hdr(pkt, &hdr);
 	if (!tcp_hdr) {
-		return NET_DROP;
+		ret = NET_DROP;
+		goto out;
```

Comment:
A goto unlock would describe the action better also here.

Reply:
> Actually my comment was wrong, the "out:" label is new. The unlock label is actually more descriptive so I will change it.

Actually, I'd say the "out" name was better: it's the difference between interface and implementation. The high-level action is "out", it's current implementation is unlocking, but maybe later it'll update some stats counters, or do sth else completely.